vic/drivers/shared_all/src/vic_time.c

```diff
@@ -851,7 +853,21 @@ time_delta(dmy_struct        *dmy_current,
             }
         }
         else if (freq == FREQ_NYEARS) {
+            // advance year
             dmy_next.year += n;
+            // re-calculate day_in_year (NOTE: currently will have problem
+            // if dmy_next.year is not leap year but date is Feb 29 !)
+            make_lastday(global_param.calendar, dmy_next.year, lastday);
+            dmy_next.day_in_year = 0;
+            for ( i = 1; i <= MONTHS_PER_YEAR; i++ ) {
```

Comment:
C-counters typically run from `0` through `< LIMIT`. That also gets around the `i-1`